torch/quantization/_equalize.py

```diff
@@ -0,0 +1,111 @@
+import torch
+import torch.nn as nn
+import copy
+
+def channel_range(input, axis=0):
+    ''' finds the range of weights associated with a specific channel
+
+    '''
+    size_of_tensor_dim = len(input.size())
+    mins = input
+    maxs = input
+
+    # reshape input to make specified axis the last axis in input
+    temp = list(range(size_of_tensor_dim))
+    temp[size_of_tensor_dim - 1] = axis
+    temp[axis] = size_of_tensor_dim - 1
+
+    mins = mins.permute(temp)
+    maxs = maxs.permute(temp)
+
+    # minimizing over all axises except the last axis
+    for i in range(size_of_tensor_dim - 1):
+        mins = torch.min(mins, 0)[0]
+        maxs = torch.max(maxs, 0)[0]
+
+    assert(mins.size()[0] == input.size()[axis])
+    return maxs - mins
+
+def cross_layer_equalization(module1, module2, output_axis=0, input_axis=1):
+    ''' Given two adjacent tensors', the weights are scaled such that
+    the ranges of the first tensors' output channel are equal to the
+    ranges of the second tensors' input channel
+
+    '''
+    if module1.weight.size()[output_axis] != module2.weight.size()[input_axis]:
```

Comment:
you still need to check the types for the modules, but the output_axis/input_axis can be fixed.

Reply:
like:
```
_supported_types = [torch.nn.Conv2d, torch.nn.Linear]
if type(module1) not in _supported_types or type(module2) not in _supported_types:
    raise ValueError("module type not supported:", type(module1), " ",type(module2))
```
you can do this here or in `equalize` function.